### PR TITLE
gc: Add `--not-in-remote`.

### PR DESCRIPTION
**Pull request by daavoo**
Per https://github.com/iterative/dvc/pull/9350

### CODE REVIEW

File: content/docs/command-reference/gc.md [hunk 1]
```diff
@@ -120,6 +121,19 @@ project we want to clear.
...
+  present in the remote.
+
+  For objects using the
+  [`remote` field](/doc/user-guide/project-structure/dvc-files#output-entries),
+  the check will be against that remote.
+
+  For objects not using the
+  [`remote` field](/doc/user-guide/project-structure/dvc-files#output-entries),
+  the check will be against the default remote unless a specific one is given
+  with `-r`.
```
Comment (dberenbaum) [inline content/docs/command-reference/gc.md:135]: Does the same apply for `-c`?

Comment (daavoo) [reply content/docs/command-reference/gc.md:135]: Updated with the same description. not sure if it should be phrased differently for the `cloud` case or not

File: content/docs/command-reference/gc.md [hunk 2]
```diff
@@ -120,9 +121,31 @@ project we want to clear.
...
+
 - `-c`, `--cloud` - remove files in remote storage in addition to local cache.
-  **This option is dangerous.** The default remote is used unless a specific one
-  is given with `-r`.
+
+  **This option is dangerous.**
+
+  For objects using the
+  [`remote` field](/doc/user-guide/project-structure/dvc-files#output-entries),
+  the check will be against that remote.
```
Comment (dberenbaum) [inline content/docs/command-reference/gc.md:143]: WDYT about adding something like this?

```
Any files not using the `remote` field may be deleted from that remote.
```

And then below something like this?

```
Any files using the `remote` field may be deleted from the default remote.
```

**Review by dberenbaum — APPROVED** [review]

**Comment by omesser** [comment]
Released in DVC [2.56.0](https://github.com/iterative/dvc/releases/tag/2.56.0)

**Comment by omesser** [comment]
🙌 thanks @daavoo !

**Pull request merged**